pkg/virt-operator/kubevirt.go

```diff
@@ -283,6 +296,27 @@ func (c *KubeVirtController) getKubeVirtKey() (string, error) {
 	return "", nil
 }
 
+func (c *KubeVirtController) sccAddHandler(obj interface{}, expecter *controller.UIDTrackingControllerExpectations) {
+	o := obj.(metav1.Object)
+	if util.IsManagedByOperator(o.GetLabels()) {
+		c.genericAddHandler(obj, expecter)
+	}
+}
+
+func (c *KubeVirtController) sccUpdateHandler(old, cur interface{}, expecter *controller.UIDTrackingControllerExpectations) {
+	o := cur.(metav1.Object)
+	if util.IsManagedByOperator(o.GetLabels()) {
+		c.genericUpdateHandler(old, cur, expecter)
+	}
+}
+
+func (c *KubeVirtController) sccDeleteHandler(obj interface{}, expecter *controller.UIDTrackingControllerExpectations) {
+	o := obj.(metav1.Object)
```

Comment:
take a look at the `genericDeleteHandler` function. You'll need to add a similar block there to handle the `cache.DeletedFinalStateUnknown` obj state.

Reply:
Hm, the thing that `cache.DeletedFinalStateUnknown` is handled in `genericDeleteHandler` is not enough? I am passing the whole SCC object to `genericDeleteHandler` ([link](https://github.com/kubevirt/kubevirt/pull/2545/files#diff-dab38d1bd4e3b7017adcaf4fc06f65e1R316)) if it's managed by the operator, so failed delete process will be reported anyway. Am I missing something here? Or maybe I should check `cache.DeletedFinalStateUnknown` state for all SCC objects (not only created by the operator)?

Reply:
The problem is with the cast of the obj and the label check that occurs before passing to genericDeleteHandler. We're not guaranteed in the delete handler that the obj is of type metav1.Object, which means the cast could fail and the label check wouldn't cause the genericDeleteHandler function to be invoked.

We have to also check the tombstone object in the delete callback.

Reply:
ah, that makes perfect sense, looking a second time with a fresh mind made the issue understandable, thanks for the explanation.

Reply:
done